### DISCUSSION TITLE
Is StackLayout the preferred base for MAUI custom layouts?

### DISCUSSION THREAD

**Discussion by Dbquity** [dotnet/maui#5283] [Q&A] [answered]
Thanks to @hartez for the illustrative examples of custom layouts at https://github.com/hartez/CustomLayoutExamples!

I plan to do a custom layout for MAUI and would like to ask:
With Horizontal- and VerticalStackLayout being part of MAUI, is StackLayout still the one to specialize for custom layouts?

I am targeting a dynamic one or two column layout where each column displays a number of "fields" each of which is a combination of a label and an input control (for instance, entry or date-picker).

**Comment by Xyncgas** [comment]
Absolute layout

**Reply by Xyncgas** [reply]
You don't have to worry about it's behavior, it's just placed where you tell it to be, it's up to you to write the algorithms like emulation of stackLayout which since you are writing your custom layout it's for you

**Comment by nogginbox** [comment]
I think that `VerticalStackLayout` and `HorizontalLayout` may just be semantically nicer versions of `StackLayout Orientation="{Horizontal/Vertical}"`.

It's going to be much more obvious at a glance if a stack layout is vertical or horizontal. I can't see that you lose or gain anything else, except you will not be able to change the orientation programmatically.

I may switch to these to make my XAML more readable.

**Comment by Dbquity** [comment, marked as answer]
```StackLayout``` it is :-)
I ended up sticking to ```StackLayout``` since it has the overridable ```CreateLayoutManager()``` method and pairs nicely with the ```StackLayoutManager``` class, which I use as a base class for my own ```FieldsLayout.Manager```.
